ic-utils/src/interfaces/management_canister/builders.rs

```diff
@@ -555,6 +567,22 @@ pub enum InstallMode {
     Upgrade(Option<CanisterUpgradeOptions>),
 }
 
+impl InstallMode {
+    /// Fills in all `Option` fields with their default values.
+    pub fn canonicalize(&self) -> InstallMode {
+        match self {
+            Self::Upgrade(u) => {
+                let mut opts = u.unwrap_or_default();
+                opts.skip_pre_upgrade.get_or_insert(false);
+                opts.wasm_memory_persistence
+                    .get_or_insert(WasmMemoryPersistence::Replace);
+                Self::Upgrade(Some(opts))
+            }
+            x => *x,
+        }
+    }
+}
+
```

Comment:
Could we keep the fields as-is without the `canonicalize()` method? I believe optional fields should retain their `None` values rather than being transformed.